db/toblock.c

```diff
@@ -5418,12 +5418,20 @@ static int toblock_main_int(struct javasp_trans_state *javasp_trans_handle,
         }
 
         if (!rowlocks) {
-            int t = time_epoch();
-            memcpy(p_buf_fstblk, &t, sizeof(int));
-            rc = bdb_blkseq_insert(thedb->bdb_env, parent_trans, bskey,
-                                   bskeylen, buf_fstblk,
-                                   p_buf_fstblk - buf_fstblk + sizeof(int),
-                                   &replay_data, &replay_len);
+            extern int gbl_always_send_cnonce;
+            // if RC_INTERNAL_RETRY && replicant_can_retry don't add to blkseq
+            if (outrc == ERR_BLOCK_FAILED && err.errcode == ERR_VERIFY && 
+                (gbl_always_send_cnonce && iq->snap_info.replicant_can_retry)) {
```

Comment:
Use iq->have_snap_info instead of gbl_always_send_cnonce for this check .. it's unlikely, but possible to toggle gbl_always_send_cnonce to false while the block processor is executing.  This scenario would allow a verify-error to be committed to blkseq.

Reply:
Make this a "review"